src/Arrays/LazyArrays.jl

```diff
@@ -193,6 +202,11 @@ function _array_cache!(dict::Dict,a::LazyArray)
   (cg, cgi, cf), IndexItemPair(index, item)
 end
 
+function invalidate_cache!(cache::Tuple{<:Any, <:IndexItemPair})
```

Comment:
Isn't this redundant with the `invalidate_cache(::Tuple)` ? I think you could just define it for `IndexItemPair`.